### ISSUE TITLE
[java] UnusedPrivateMethod false-positive

### ISSUE THREAD

**Issue by Wolf2323** [pmd/pmd#5664]
**Affects PMD Version:**  7.11.0 (also tested with 7.12.0)

**Rule:** [UnusedPrivateMethod](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**
A private method in a nested class get detected as unused since newer versions. The last one where it was working was 7.10.0

**Code Sample demonstrating the issue:**

```java
@ExtendWith(MockitoExtension.class)
class TimeProfilerTests {
	@Nested
	class operations_test {
		private final long[] offset = {0L};

		private static Long offsetTimeSupplier(final long[] offset) {
			return offset[0];
		}

		@BeforeEach
		void setUp() {
			offset[0] = offsetTimeSupplier(offset);
		}
	}
```

**Expected outcome:**
The method is used in the setup method, so it should have the violation.

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *Maven*
